server/migrations/20240525041559-addCollectionJobSecret.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const uuid = require('uuid');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    return queryInterface.sequelize.transaction(async transaction => {
+      await queryInterface.addColumn(
+        'CollectionJob',
+        'secret',
+        {
+          type: Sequelize.DataTypes.UUID,
+          allowNull: false,
+          defaultValue: uuid.NIL
```

Comment:
Given how this is handled with the `beforeValidate` hook in `models/CollectionJob`, should this be `uuid()` as well?

Reply:
As mentioned in the other comment in this file, putting a `uuid()` here would make it the same UUID for all entries, I'd rather have that "it never got set" value to stand out with uuid.NIL.

If something was insterting into the database without the `beforeValidate` hook `00000-0000-00-00-0` (yeah didn't feel like matching the uuid spec) would still work for the key anyway, but something else is already probably very broken.

Reply:
Ah okay, understood!